MigrationExecutorApp/ChangeFeedProcessortHost.cs

```diff
@@ -108,92 +121,131 @@ public async Task CloseAsync()
             Trace.TraceInformation("Host name {0}", hostName);
 
-            // monitored collection info 
-            var documentCollectionLocation = new DocumentCollectionInfo
-            {
-                Uri = new Uri(this.config.MonitoredUri),
-                MasterKey = this.config.MonitoredSecretKey,
-                DatabaseName = this.config.MonitoredDbName,
-                CollectionName = this.config.MonitoredCollectionName
-            };
 
-            var policy = new ConnectionPolicy()
+            var docTransformer = new DefaultDocumentTransformer();
+          
+            if (!String.IsNullOrEmpty(config.BlobConnectionString))
             {
-                ConnectionMode = ConnectionMode.Direct,
-                ConnectionProtocol = Protocol.Tcp
-            };
-
-            policy.PreferredLocations.Add("North Europe");
+                BlobServiceClient blobServiceClient = new BlobServiceClient(config.BlobConnectionString);
+                containerClient = blobServiceClient.GetBlobContainerClient(config.BlobContainerName);
+                await containerClient.CreateIfNotExistsAsync();
+            }
 
-            // lease collection info 
-            var leaseCollectionLocation = new DocumentCollectionInfo
+            DateTime starttime = DateTime.MinValue.ToUniversalTime();
+            if (config.DataAgeInHours.HasValue)
             {
-                Uri = new Uri(this.config.LeaseUri),
-                MasterKey = this.config.LeaseSecretKey,
-                DatabaseName = this.config.LeaseDbName,
-                CollectionName = this.config.LeaseCollectionName,
-                ConnectionPolicy = policy
-            };
-
-            // destination collection info 
-            var destCollInfo = new DocumentCollectionInfo
-            {
-                Uri = new Uri(this.config.DestUri),
-                MasterKey = this.config.DestSecretKey,
-                DatabaseName = this.config.DestDbName,
-                CollectionName = this.config.DestCollectionName
-            };
-
-            var processorOptions = new ChangeFeedProcessorOptions();
-            if(config.DataAgeInHours.HasValue) {
                 if (config.DataAgeInHours.Value >= 0)
                 {
-                    processorOptions.StartTime = DateTime.UtcNow.Subtract(TimeSpan.FromHours(config.DataAgeInHours.Value));
+                    starttime = DateTime.UtcNow.Subtract(TimeSpan.FromHours(config.DataAgeInHours.Value));
                 }
-            } else
-            {
-                processorOptions.StartFromBeginning = true;
             }
 
-            processorOptions.LeaseRenewInterval = TimeSpan.FromSeconds(30);
+            changeFeedProcessor = sourceCollectionClient.GetContainer(config.MonitoredDbName, config.MonitoredCollectionName)
+                .GetChangeFeedProcessorBuilder<Document>("Live Data Migrator", ProcessChangesAsync)
+                .WithInstanceName("consoleHost")
```

Comment:
Is this expected to run on a single machine?

Reply:
@revinjchalil - can you comment on this?

Reply:
It is expected to run on multiple instances with 5 being the default number of instances. Whenever a new instance gets added during scale-up, it starts lease partitions to process. Max number of instances possible would be the number of partitions in the source collection.

Reply:
@TheovanKraay , use "hostname" Guid from above instead of "consoleHost" so that hostnames remain unique and not static for the distributed processing?

Reply:
Thanks @revinjchalil I have changed in last update.